### ISSUE TITLE
In newest Obsidian update, plugin gets into infinite error loop and then crashes Obsidian

### ISSUE THREAD

**Issue by spencerc99** [delashum/obsidian-checklist-plugin#158]
Seeing this error repeat over and over in console until Obsidian crashes :/ on the newest 1.0 version
![image](https://user-images.githubusercontent.com/14796580/199606975-7c0722ee-7536-46dd-8cf5-d57603e953ef.png)

**Comment by user873475320** [comment]
I have the same :(

**Comment by wbsu2003** [comment]
I have the same problem in v1.03

**Comment by flankstaek** [comment]
I am not able to replicate this issue with v1.0.3 of Obsidian and v2.2.10 of this plugin. Could those of you having the issue please verify the version of Obsidian & this plugin you are using and your operating system to help figure out the cause?

**Comment by user873475320** [comment]
> I am not able to replicate this issue with v1.0.3 of Obsidian and v2.2.10 of this plugin. Could those of you having the issue please verify the version of Obsidian & this plugin you are using and your operating system to help figure out the cause?

Hello and thank you so much, I have Kubuntu 22.04.1 LTS x86_64, Obsidian v1.0.3 and v2.2.10 of the plugin. Now I tried enable and use the plugin v 2.2.12 but I got the same(

**Comment by Verayth** [comment]
I get a similar error message starting obsidian on Windows 10 with this enabled.  It usually caused obsidian to blackscreen while re-indexing at about 60 of 1038, but doesn't while the console window is open for some reason.  Also, the re-index runs VERY slowly, probably due to this error.  I didn't have the patience to wait for it to finish, and marking the plugin disabled in the options causes the errors to stop.

```
VM185 plugin:obsidian-checklist-plugin:6410 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'filePath')
    at eval (VM185 plugin:obsidian-checklist-plugin:6410:77)
    at md.renderer.rules.<computed> [as regexp-6215] (VM185 plugin:obsidian-checklist-plugin:6382:14)
    at Renderer.renderInline (VM185 plugin:obsidian-checklist-plugin:1022:32)
    at Renderer.render (VM185 plugin:obsidian-checklist-plugin:1047:26)
    at MarkdownIt.render (VM185 plugin:obsidian-checklist-plugin:4908:28)
    at formTodo (VM185 plugin:obsidian-checklist-plugin:6544:17)
    at findAllTodosInFile (VM185 plugin:obsidian-checklist-plugin:6493:18)
    at eval (VM185 plugin:obsidian-checklist-plugin:6452:17)
    at Generator.next (<anonymous>)
    at fulfilled (VM185 plugin:obsidian-checklist-plugin:50:24)

```

**Comment by delashum** [comment]
Fixed in 2.2.13
